lib/ecto/schema.ex

```diff
@@ -552,6 +555,11 @@ defmodule Ecto.Schema do
         assocs = @ecto_assocs |> Enum.reverse
         embeds = @ecto_embeds |> Enum.reverse
         loaded = Ecto.Schema.__loaded__(__MODULE__, @struct_fields)
+        redacted_fields = if @ecto_redact_password do
+          [:password | @ecto_redacted_fields || []]
```

Comment:
`@ecto_redacted_fields` should always return an empty list since it is marked as accumulate, so you can skip the ` || []` parts. :)